src/libutil/tests/url-name.cc

```diff
@@ -0,0 +1,67 @@
+#include "url-name.hh"
+#include <gtest/gtest.h>
+
+namespace nix {
+
+/* ----------- tests for url-name.hh --------------------------------------------------*/
+
+    TEST(getNameFromURL, getsNameFromURL) {
+        ASSERT_EQ(getNameFromURL(parseURL("path:/home/user/project")), "project");
+        ASSERT_EQ(getNameFromURL(parseURL("path:~/repos/nixpkgs#packages.x86_64-linux.hello")), "hello");
+        ASSERT_EQ(getNameFromURL(parseURL("path:.#nonStandardAttr.mylaptop")), "nonStandardAttr.mylaptop");
+        ASSERT_EQ(getNameFromURL(parseURL("path:./repos/myflake#nonStandardAttr.mylaptop")), "nonStandardAttr.mylaptop");
+        ASSERT_EQ(getNameFromURL(parseURL("path:./nixpkgs#packages.x86_64-linux.complex^bin,man")), "complex");
+        ASSERT_EQ(getNameFromURL(parseURL("path:./myproj#packages.x86_64-linux.default^*")), "myproj");
```

Comment:
Nice going on testing the explicit outputs as well :+1:.

Reply:
Thank you! And wouldn't you know it, adding those two tests actually uncovered a problem that had to be fixed.